### DISCUSSION TITLE
How to use “MP_QSTR_y”？

### DISCUSSION THREAD

**Discussion by Mars-CN** [micropython/micropython#12107] [ESP32] [answered]
I encountered a problem when using C to extend MicroPython. There is a class with a member function called 'y', but I encountered an error when adding members to this class.

/home/mars/esp/micropython/ports/facemoo/build/frozen_content.c:317:5: error: redeclaration of enumerator 'MP_QSTR_y'
     MP_QSTR_y,
     ^~~~~~~~~
In file included from /home/mars/esp/micropython/py/obj.h:33,
                 from /home/mars/esp/micropython/py/objint.h:30,
                 from /home/mars/esp/micropython/ports/facemoo/build/frozen_content.c:16:
/home/mars/esp/micropython/ports/facemoo/build/genhdr/qstrdefs.generated.h:1002:6: note: previous definition of 'MP_QSTR_y' was here
 QDEF(MP_QSTR_y, 46556, 1, "y")
      ^~~~~~~~~
/home/mars/esp/micropython/py/qstr.h:41:34: note: in definition of macro 'QDEF'
 #define QDEF(id, hash, len, str) id,

Where is the problem?

**Comment by robert-hh** [comment]
Try to remove all previous build files (`rm -r build*`) and restart the build. You can also use `make BOARD=xxx clean`, but that would also remove the build of the IDF files and then a new build takes longer.

**Reply by jimmo** [reply, marked as answer]
Yep a `make clean` will solve this.

The build can handle the set of known qstrs changing and rebuilds as necessary, except in the case where a qstr that was previously only in the frozen set is now also in the base set.
